### ISSUE TITLE
Final def in trait leads to java.lang.VerifyError: class L overrides final method c.(I)I

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#5094]
Same bug as https://github.com/scala/bug/issues/11136:
```scala
trait IO {
  def c(x: Int): Int = ???
}
trait SO extends IO {
  override final def c(x: Int): Int = ???
}
trait SOIO extends IO {
  override def c(x: Int): Int = ???
}
trait SOSO extends SOIO with SO
abstract class AS extends SO
class L extends AS with SOSO
object Test {
  def main(args: Array[String]): Unit = {
    new L
  }
}
```
```scala
Exception in thread "main" java.lang.VerifyError: class L overrides final method c.(I)I
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at Test$.main(i11136.scala:16)
        at Test.main(i11136.scala)
```
Decompilation output:
```java
% cfr AS
public abstract class AS
implements SO {
    public AS() {
    }

    @Override
    public final int c(int x) {
        return SO.super.c(x);
    }
}
```
```java
% cfr L
public class L
extends AS
implements SOIO,
SOSO {
    public L() {
    }

    @Override
    public int c(int x) {
        return SOIO.super.c(x);
    }
}
```

**Comment by smarter** [comment]
A fix has now been merged in 2.13 thanks to @adriaanm heroic efforts: https://github.com/scala/scala/pull/7439, we should port this over.

**Comment by nicolasstucki** [comment]
Now it only has

```java
> java -jar cfr-0.148.jar AS.class 
public abstract class AS
implements SO {
}
> java -jar cfr-0.148.jar L.class 
public class L
extends AS
implements SOIO,
SOSO {
}
```
